### PR TITLE
Ensure that dev caches can be found by pull requests

### PR DESCRIPTION
**Pull request by jnsiemer**

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -84,7 +84,6 @@ jobs:
       
       - name: Archive release build
         uses: actions/upload-artifact@v1
-        if: github.ref == 'ref/head/master'
         with:
           name: ${{ env.PROJECT_NAME }}-release_build-v${{ env.PROJECT_VERSION }}
           path: target/release/${{ env.PROJECT_NAME }}
```
Comment (Marvin-Beckmann) [inline .github/workflows/main.yml:89]: wouldn't this mean that after every push on dev a release version is build?

File: .github/workflows/pull_request.yml [hunk 2]
```diff
@@ -1,7 +1,13 @@
...
 # consistency regarding formatting and idiomatic Rust
 
-on: [ pull_request ]
+on: 
+  push:
+    branches:
+      - dev
+  pull_request:
+    branches:
+      - "**"
```
Comment (Marvin-Beckmann) [inline .github/workflows/pull_request.yml:10]: would this mean that the cash is used for every single branch? Even if we lets say realize that we have to reset the branch to an older state, would this mean that the cache would still be used for this PR?

**Review by Marvin-Beckmann — CHANGES_REQUESTED** [review]
I am not really shure why this fixes our issue, can you provide an intuition why it does. Please add it to the description of the PR

File: .github/workflows/main.yml (hunk 1 above)
Comment (MoogSven) [reply .github/workflows/main.yml:89]: no, because it is now just running on main (first change)

**Review by MoogSven — APPROVED** [review]
looks good :+1:

**Pull request merged**